### ISSUE TITLE
The reported addresses of mispredicted branches are off by a few instructions

### ISSUE THREAD

**Issue by OleksiiOleksenko** [OleksiiOleksenko/SpecFuzz#9]
As @enlighten5 mentioned in #7 , when SpecFuzz reports a detected speculative overflow, the addresses of mispredicted branches are a bit off. Specifically, SpecFuzz reports not the address of the branch, but an address within the instrumentation that we add to this branch. It does not really influence the correctness of the approach - it's still the same location in the program. But it could be confusing when manually analysing the results.

**Comment by OleksiiOleksenko** [comment]
Unfortunately, this would have to stay as is.

Current instrumentation of conditional branches looks like this:
```asm
1.  mov %rsp, current_rsp  // preserve stack pointer
2.  callq <specfuzz_chkp>
3.  mov current_rsp,%rsp // restore stack pointer
4.  jmp  .original_branch  // skip simulation after rollback
5.  ja .target  // inverse to original jump (simulation of misprediction)
6.  jmp .fall_through
7.  .original_branch: jbe .target
8.  .fall_through: ...
```

SpecFuzz reports line 3 as an address of the mispredicted branch. This is, of course, confusing - the actual mispredicted branch is line 7 or, in worst case, line 5. However, the only address available to the specfuzz runtime is the return address of `specfuzz_chkp` and therefore, we report it. 

It might seem that calculating the address of line 7 from the address of line 3 is as simple as adding a fixed offset. It's more complicated, though. The size of of instructions at lines 5 and 6 depends on how far their jump targets are located. This information is not available at runtime and getting during the compilation is also tricky. Moreover, propagating this info to runtime would further complicate the instrumentation, which is undesirable.

Bottom line, I did not figure out how to fix this issue without sacrificing the fuzzing performance. And since it's a usability rather than correctness issue, I decided to leave it as it is.
